### PR TITLE
Updating authentication file with bearer authentication

### PR DESCRIPTION
**Pull request by vad33r**
Add support to django-oauth2-provider and django-oauth-toolkit

### CODE REVIEW

File: openedx/core/lib/api/authentication.py [hunk 1]
```diff
@@ -105,3 +136,28 @@ def _get_dot_token(self, access_token):
...
+
+    But until then, this authentication class ensures that the user is logged in,
+    but does not require that their account "is_active".
+
+    This class can be used for an OAuth2-accessible endpoint that allows users to access
+    that endpoint without having their email verified.  For example, this is used
+    for mobile endpoints.
+    """
+
+    allow_inactive_users = True
```
Comment (andrey-canon) [inline openedx/core/lib/api/authentication.py:163]: the last empty line is missing, Why do you include this line? where `allow_inactive_users` is used ?

File: package-lock.json [hunk 2]
```diff
@@ -254,7 +254,7 @@
     "abbrev": {
       "version": "1.1.1",
       "resolved": "https://registry.npmjs.org/abbrev/-/abbrev-1.1.1.tgz",
-      "integrity": "sha512-nne9/IiQ/hzIhY6pdDnbBtz7DjPTKrY00P/zvPSm5pOFkl6xuGrGnXn/VtTNNfNtAfZ9/1RtehkszU9qcTii0Q=="
+      "integrity": "sha1-+PLIh60Qv2f2NPAFtph/7TF5qsg="
```
Comment (andrey-canon) [inline package-lock.json:257]: This file must not be included in this pr

**Review by andrey-canon — CHANGES_REQUESTED** [review]

File: openedx/core/lib/api/authentication.py (hunk 1 above)
Comment (mariajgrimaldi) [reply openedx/core/lib/api/authentication.py:163]: is this class used?

**Review by felipemontoya — APPROVED** [review]
Lets fix the test and write the migration path document somewhere, but I'll leave the approval already

**Pull request merged**